components/pluralrules/src/operands.rs

```diff
@@ -0,0 +1,186 @@
+//! Plural operands in compliance with [CLDR Plural Rules](http://unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules).
+//!
+//! See [full operands description](http://unicode.org/reports/tr35/tr35-numbers.html#Operands).
+//!
+//! # Examples
+//!
+//! From int
+//!
+//! ```
+//! use std::convert::TryFrom;
+//! use icu_pluralrules::operands::*;
+//! assert_eq!(PluralOperands {
+//!    n: 2_f64,
+//!    i: 2,
+//!    v: 0,
+//!    w: 0,
+//!    f: 0,
+//!    t: 0,
+//! }, PluralOperands::from(2 as usize))
+//! ```
+//!
+//! From float
+//!
+//! ```
+//! use std::convert::TryFrom;
+//! use icu_pluralrules::operands::*;
+//! assert_eq!(Ok(PluralOperands {
+//!    n: 1234.567_f64,
+//!    i: 1234,
+//!    v: 3,
+//!    w: 3,
+//!    f: 567,
+//!    t: 567,
+//! }), PluralOperands::try_from("-1234.567"))
+//! ```
+//!
+//! From &str
+//!
+//! ```
+//! use std::convert::TryFrom;
+//! use icu_pluralrules::operands::*;
+//! assert_eq!(Ok(PluralOperands {
+//!    n: 123.45_f64,
+//!    i: 123,
+//!    v: 2,
+//!    w: 2,
+//!    f: 45,
+//!    t: 45,
+//! }), PluralOperands::try_from(123.45))
+//! ```
+#![cfg_attr(feature = "cargo-clippy", allow(clippy::cast_lossless))]
+use std::convert::TryFrom;
+use std::isize;
+use std::str::FromStr;
+
+/// A full plural operands representation of a number. See [CLDR Plural Rules](http://unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules) for complete operands description.
+#[derive(Debug, PartialEq)]
+pub struct PluralOperands {
+    /// Absolute value of input
+    pub n: f64,
```

Comment:
I'm reluctant to change names of those fields because they are deeply ingrained in how operands are used by the rules. The canonical use is a rule like `i = 0 and v = 1` and if I "renamed" them here to something like `Operands::Number`, `Operands::Integer` and `Operands::VisibleFractionDigits` then we'd have to carry the rule as `Integer = 0 and VisibleFractionDigits = 1`.

That may cost us when trying to encode the rules as static Rust functions.

I'll experiment with that before I push the patch for review.